Source/EmbeddedBoundary/WarpXFaceExtensions.cpp

```diff
@@ -658,6 +682,38 @@ WarpX::ComputeEightWaysExtensions() {
 #endif
 }
 
+void
+WarpX::ApplyBCKCorrection(const int idim) {
+#if defined(AMREX_USE_EB) and !defined(WARPX_DIM_RZ)
+    auto const &cell_size = CellSize(maxLevel());
+
+    const amrex::Real dx = cell_size[0];
+    const amrex::Real dy = cell_size[1];
+    const amrex::Real dz = cell_size[2];
+
+    for (amrex::MFIter mfi(*Bfield_fp[maxLevel()][idim]); mfi.isValid(); ++mfi) {
+
+        amrex::Box const &box = mfi.validbox();
+        auto const &flag_ext_face = m_flag_ext_face[maxLevel()][idim]->array(mfi);
+        auto const &flag_info_face = m_flag_info_face[maxLevel()][idim]->array(mfi);
+        auto const &S = m_face_areas[maxLevel()][idim]->array(mfi);
+        auto const &lx = m_face_areas[maxLevel()][0]->array(mfi);
+        auto const &ly = m_face_areas[maxLevel()][1]->array(mfi);
+        auto const &lz = m_face_areas[maxLevel()][2]->array(mfi);
```

Comment:
Would it be possible to replace these `auto` as well? Something like `amrex::Array4<amrex::Real>` would probably work.